src/app/ReadClient.h

```diff
@@ -255,6 +261,22 @@ class ReadClient : public Messaging::ExchangeDelegate
     CHIP_ERROR AbortExistingExchangeContext();
     const char * GetStateStr() const;
 
+    /**
+     * Validate that the Event ID and Cluster ID in the header match that of the type information present in the object before
+     * decoding the data. The template parameter T is generally expected to be a ClusterName::Events::EventName::Type struct
+     *
+     * @param [in] aEventHeader  The header of the event being validated.
+     * @param [in] aEvent         The event data.
+     */
+    template <typename EventDataT>
+    CHIP_ERROR CheckEventHeaderWithClusterObject(const EventHeader & aEventHeader, const EventDataT & aEvent)
```

Comment:
Why just checking? Why not make this do decode as well?

Reply:
updated to decode as well